src/Stack/Application.php

```diff
@@ -97,8 +100,31 @@ protected function buildResponseFromResponseBag(ResponseBag $responseBag)
         $result = $responseBag->getResult();
         $resultCode = $responseBag->getResultCode();
         $headers = $responseBag->getHeaders();
+        $type = $responseBag->getType();
 
-        return new Response($result, $resultCode, $headers);
+        if ($type === ResponseBag::TYPE_DEFAULT) {
```

Comment:
Do we really need a factory when we don't have dependency resolution problematic (i'm not against but for 3 types and no real problematic behind)